src/Tools/ExternalAccess/Razor/RazorAnalyzerAssemblyResolver.cs

```diff
@@ -1,58 +1,50 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
+#if NET
 
 using System;
-using System.Collections.Generic;
 using System.Composition;
-using System.Diagnostics;
+using System.IO;
+using System.Linq;
 using System.Reflection;
+using System.Runtime.Loader;
 using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.VisualStudio.Composition;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
 {
     [Export(typeof(IAnalyzerAssemblyResolver)), Shared]
     [method: ImportingConstructor]
     [method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    internal class RazorAnalyzerAssemblyResolver() : IAnalyzerAssemblyResolver
+    internal sealed class RazorAnalyzerAssemblyResolver() : IAnalyzerAssemblyResolver
     {
-        private static Func<AssemblyName, Assembly?>? s_assemblyResolver;
+        private const string RazorCompilerAssemblyName = "Microsoft.CodeAnalysis.Razor.Compiler";
+        private const string RazorUtilsAssemblyName = "Microsoft.AspNetCore.Razor.Utilities.Shared";
+        private const string ObjectPoolAssemblyName = "Microsoft.Extensions.ObjectPool";
```

Comment:
Where in the code do we keep the list of generators that the IDE remaps from SDK to the VS copy? Wanted to make sure there is agreement on all the assemblies in that map and what we redirect here.